### PR TITLE
Issue #13109: Kill mutation for PackageDeclarationCheck

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13109: Kill mutation for PackageDeclarationCheck

-------------

# Mutation
https://github.com/checkstyle/checkstyle/blob/81448ea434cda883b748d3d2c77b2f58a8b22bbe/config/pitest-suppressions/pitest-coding-1-suppressions.xml#L48-L55

### CODE REVIEW

**Comment by romani** [comment]
why draft ? no regression report is executed.

we already discussed somewhere that all reset in beginTree are for case when same instance of Check is used on two files, Check is state-ful, and we should not affect new files by values in fields from previous java file.

**Comment by Kevin222004** [comment]
@romani 
Regression https://kevin222004.github.io/reports/p1/2023-06-02-T-08-07-37/test-report/index.html

**Comment by romani** [comment]
How long it takes for your local to generate such report ?
Do you able to do something else meanwhile ?

**Comment by Kevin222004** [comment]
@romani i am not generating report on local 
I generate using https://github.com/Kevin222004/Kevin222004.github.io

**Comment by Kevin222004** [comment]
@romani https://kevin222004.github.io/Regression/

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java [hunk 1]
```diff
@@ -145,11 +145,6 @@ public int[] getAcceptableTokens() {
         return getRequiredTokens();
     }
 
-    @Override
-    public void beginTree(DetailAST ast) {
-        defined = false;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java:150]: reminder, all initialization in this method, is reset of Check state before starting new file, to reproduce problem we need to run same Check instance on 2 special input files to let "defined = true" affect validation in second file even it has no java code that make "defined = true".

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java:150]: Do not deal with pure Junit method , we should cover it by Inputs.
New or existing verify method that takes creates check ones and execute two files sequentially.

Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java]: @romani, I have tried to run 2 files sequentially but the only thing changed here is if I make https://github.com/checkstyle/checkstyle/blob/24a4b06182f9b9066a684715f324d2bbf7099d10/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java#L148-L151 
defined = true then my test case will fail but it is still not resolving our actual mutation which is to remove that line 
`defined = false`

Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java]: Done

**Review by romani — APPROVED** [review]
Ok to merge.
Just Input based test extension, best scenario.

I move it further for review by others to learn on this.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/packagedeclaration/InputPackageDeclarationCheck.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.packagedeclaration;
```
Comment (rdiachenko) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/packagedeclaration/InputPackageDeclarationCheck.java:1]: So, why do we need this new input file if it is not used in the new test?

**Review by rdiachenko — APPROVED** [review]
lgtm

**Review by Vyom-Yadav — APPROVED** [review]
LGTM!

**Pull request merged**